ads/google/a4a/utils.js

```diff
@@ -113,9 +114,27 @@ export function googleAdUrl(
     const viewportRect = viewport.getRect();
     const iframeDepth = iframeNestingDepth(win);
     const viewportSize = viewport.getSize();
-    if (ValidAdContainerTypes.indexOf(adElement.parentElement.tagName) >= 0) {
-      queryParams.push({name: 'amp_ct',
-                        value: adElement.parentElement.tagName});
+
+    // Detect container types.
+    let parentElement = adElement.parentElement;
+    let tagName = parentElement.tagName.toUpperCase();
+    const containerTypeSet = {};
+    while (parentElement && ValidAdContainerTypes[tagName]) {
+      containerTypeSet[ValidAdContainerTypes[tagName]] = true;
+      parentElement = parentElement.parentElement;
+      tagName = parentElement.tagName.toUpperCase();
+    }
+    const containerTypeList = [];
```

Comment:
queryParams.push({name: 'act', value: Object.keys(containerTypeSet).join()});

This will put value empty string if no container types, not sure if that's a problem and/or it will not be included in your request.